product_supplierinfo_discount/models/purchase_order.py

```diff
@@ -51,7 +51,10 @@ def onchange_product_id(
              ('name', 'child_of', partner.commercial_partner_id.id)])
         if not sinfos:
             return res
-        seller_uom = sinfos.product_uom.id or False
+        if len(sinfos) > 1:
```

Comment:
Please change this to `seller_uom = sinfos[:1].product_uom.id`. It makes the same and it doesn't fail on empty values (not on v8).